pkg/util/rand/rand.go

```diff
@@ -54,3 +54,11 @@ func Seed(seed int64) {
 
 	rng.rand = rand.New(rand.NewSource(seed))
 }
+
+// Perm returns, as a slice of n ints, a pseudo-random permutation of the integers [0,n)
+// from the default Source.
+func Perm(n int) []int {
+	rng.Lock()
```

Comment:
wouldn't this be enough ? https://github.com/ArtfulCoder/kubernetes/blob/shufflestrings/pkg/util/rand/rand.go#L32